opm/simulators/linalg/gpuistl/GpuBuffer.cpp

```diff
@@ -33,6 +33,14 @@ GpuBuffer<T>::GpuBuffer(const std::vector<T>& data)
 {
 }
 
+// some classes need a default constructor...
+// should we reintroduce the default constructor that allocates 0 bytes and initializes the handles?
```

Comment:
There is already a declared default constructor, but it seems it is not actually implemented before this, which is weird.

That aside, I think it should not allocate anything. Default construct then resize should allocate only once. I suggest removing this implementation, defaulting `m_numberOfElements` to zero in the header file, and then just defaulting the constructor in the header:
```
    /**
     * @brief Default constructor that will initialize cublas and allocate 0 bytes of memory
     */
    GpuBuffer() = default;
```
That way you do not need to change the doc string either to say 1 instead 0...

Finally, note that it is currently declared explicit in the header, which is unnecessary and misleading.

Reply:
fixed now with updated doc string and logic in resize function accounting for original size being 0